### DISCUSSION TITLE
[NEW] Product imports/updates greatly improved - Enable new feature now!

### DISCUSSION THREAD

**Discussion by EricBusch** [datafeedr/wordpress-plugins#5] [Announcements]
### Update 2022-03-07 11:01:48

This is no longer necessary as the feature has now been integrated into the [Datafeedr WooCommerce Importer plugin](https://datafeedr.me/dfrpswc) as of version 1.3.0.

If you are experiencing problems with version 1.3.0 of the Datafeedr WooCommerce Importer plugin, you can revert the changes by adding the following code to your [Custom Plugin](https://datafeedrapi.helpscoutdocs.com/article/32-creating-your-own-custom-plugin).

```php
add_filter( 'dfrpswc_enable_product_update_handler_feature_flag', '__return_false' );
```

---

Hi All,

In October, 2020 we released a new feature to the [Datafeedr WooCommerce Importer plugin](https://datafeedr.me/dfrpswc) but the new feature is _inactive_ by default.

### What does the new code do?

The new code takes advantage of native WooCommerce APIs to import and update products. This has many advantages:

- faster product imports and updates
- better handling of "counts" like product count, term counts, etc...
- better handling of sorting like "Sort by price ascending", etc...
- better attribute filtering (ie. colors, brands, etc...)

### Why is it disabled by default?

By default, the new "updater" code is disabled and you must opt-in to use it. This is because the new code could result in some of your custom code (specifically related to importing product attributes) not working as expected.

However, we have tested the new "updater" code against most of the custom code samples in our [documentation](https://datafeedr.me/docs) and did not have any problems. The custom code continued to work exactly as expected.

### What should I do?

Add the following PHP Code to your [Custom Plugin](https://datafeedrapi.helpscoutdocs.com/article/32-creating-your-own-custom-plugin).

```php
add_filter( 'dfrpswc_enable_product_update_handler_feature_flag', '__return_true' );
```

### When will this new code by enabled by default?

We are not sure yet... 

### Questions?

Sure! Please post your questions regarding this new feature below.

**Comment by EricBusch** [comment]
There have been a few custom code snippets that have been affected by these changes. Please also make sure you have updated your custom code if you have implemented the following:

- [Import additional product images as gallery images](https://datafeedrapi.helpscoutdocs.com/article/204-import-additional-product-images-as-gallery-images)
- [Import new products as drafts](https://datafeedrapi.helpscoutdocs.com/article/188-import-new-products-as-drafts)

**Reply by fiversen-hofor** [reply]
Hi. 
Should the code be functioning now. 
I'm a little bit confused because I have seen the message for very long time telling me that you're going to implement it. I don't think I have seen any improvement in the update speed on my sites and I have implemented the "add_filter...." in MyCustomCode.
kind Regards
Flemming

**Reply by EricBusch** [reply]
@fiversen-hofor Update speed could vary depending on the size of your store. But we do still recommend implementing the code above so that any custom changes you make to your site won't be affected when we turn this new feature "ON" for everyone.
